perfkitbenchmarker/linux_benchmarks/cloudsuite_dataserving_benchmark.py

```diff
@@ -0,0 +1,219 @@
+# Copyright 2015 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Runs the dataserving benchmark of Cloudsuite.
+
+More info: http://cloudsuite.ch/dataserving/
+"""
+
+import re
+
+from perfkitbenchmarker import configs
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import sample
+
+FLAGS = flags.FLAGS
+
+BENCHMARK_NAME = 'cloudsuite_dataserving'
+BENCHMARK_CONFIG = """
+cloudsuite_dataserving:
+  description: >
+      Run YCSB against Cassandra. Specify the
+      Cassandra cluster size with --num_vms. Specify the number
+      of YCSB VMs with --ycsb_client_vms.
+  vm_groups:
+    servers:
+      vm_spec: *default_single_core
+      vm_count: 1
+    clients:
+      vm_spec: *default_single_core
+      vm_count: 1
+"""
+
+
+def GetConfig(user_config):
+  config = configs.LoadConfig(BENCHMARK_CONFIG, user_config, BENCHMARK_NAME)
+  return config
+
+
+def _HasDocker(vm):
+  resp, _ = vm.RemoteCommand('command -v docker',
+                             ignore_failure=True,
+                             suppress_warning=True)
+  if resp.rstrip() == "":
+    return False
+  else:
+    return True
+
+
+def Prepare(benchmark_spec):
+  """Install docker. Pull the required images from DockerHub.
+        Create a table and load the dataset.
+
+  Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+  """
+  servers = benchmark_spec.vm_groups['servers']
+  clients = benchmark_spec.vm_groups['clients']
+
+  for vm in servers:
+    if not _HasDocker(vm):
+      vm.Install('docker')
+    vm.RemoteCommand('sudo docker pull cloudsuite/data-serving:server')
+    vm.RemoteCommand('sudo docker run -d --name cassandra-server --net host '
+                     'cloudsuite/data-serving:server')
+
+  for vm in clients:
+    if not _HasDocker(vm):
+      vm.Install('docker')
+    vm.RemoteCommand('sudo docker pull cloudsuite/data-serving:client')
+
+
+def Run(benchmark_spec):
+  """Run the dataserving benchmark.
+
+  Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+
+  Returns:
+    A list of sample.Sample objects.
+  """
+  client = benchmark_spec.vm_groups['clients'][0]
+  server = benchmark_spec.vm_groups['servers'][0]
```

Comment:
I'm not familiar with the exact steps in this benchmark, but is it intentional to only use the first VM from each group? I didn't see any special treatment for the first VM in the Prepare function.